[App Service] `az webapp up`: Add ability to deploy Linux and Windows webapps to the same resource group

StrawnSC opened the pull request:
**Description**<!--Mandatory-->
<!--Why this PR? What is changed? What is the effect? etc. A high-quality description can accelerate the review process.-->
Removes the checks that prevent deploying linux and windows webapps to the same resource group with `az webapp up`

**Testing Guide**
<!--Example commands with explanations.-->
```
# create a resource group 
az group create <rg_name>

# go to a (supported) webapp source directory 
cd <webapp_dir> 

# deploy a windows app 
az webapp up -n <app1_name> -g <rg_name>  --os-type Windows

# create a linux appservice plan 
az appservice create -g <rg_name> --name <plan_name> --is-linux

# deploy a linux app to the same resource group 
az webapp up -n <app2_name> -g <rg_name>  --os-type Linux --plan <plan_name>

```


**History Notes**
<!--If your PR is not customer-facing, use {Component Name} in the PR title. Otherwise, use [Component Name] to allow our pipeline to add the title as a history note. If you need multiple history notes or would like to overwrite the note from the PR title, please fill in the following templates.-->

N/A

<!--[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.
[Component Name 2] az command b: Add some customer-facing feature.-->

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [X] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

panchagnula commented on src/azure-cli/azure/cli/command_modules/appservice/_create_util.py:
```diff
@@ -312,12 +312,9 @@ def set_location(cmd, sku, location):
-# check if the RG value to use already exists and follows the OS requirements or new RG to be created
-def should_create_new_rg(cmd, rg_name, is_linux):
-    if (_check_resource_group_exists(cmd, rg_name) and
-            _check_resource_group_supports_os(cmd, rg_name, is_linux)):
```
Can we remove the helper method code as well - if this is not used anywhere else?

StrawnSC replied:
Just removed the function

StrawnSC commented:
> @StrawnSC nice! Can we re-run the webapp up test recordings, and add a new test if necessary?

@calvinsID I re-recorded the `az webapp up` tests and added a test that deploys a linux and windows webapp with `az webapp up` to the same resource group. I fixed the tests that were out of date

zhoxing-ms commented on src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_webapp_up_commands.py:
```diff
@@ -905,5 +905,100 @@ def test_webapp_up_generate_default_name(self, resource_group):
         shutil.rmtree(temp_dir)
 
 
+    @live_only()
```
I'd like to ask why this test is ` @live_only()`?

StrawnSC replied:
The test case was modeled after a pre-existing test function which tested the ability to use `az webapp up` to deploy a python webapp (linux). I modified it to also deploy a windows app to the same resource group.  The original test function, as well as most of the other `az webapp up` tests, are live only as well -- perhaps that's not a solid rationale though

@calvinsID @panchagnula do you know why most of these tests are live only and have any thoughts on whether they should be or not?

panchagnula replied:
@zhoxing-ms the az webapp up is a not idempotent like az webapp create i.e a simple flow is
1. webapp doesn't exist, we create the webapp, ASP, RG & deploy the code to the webapp
2. App exists, the code path is to deploy contents
 az webapp up was done as a quick start command, for users who are completely new to azure & also to have a command with minimal parameters. Hope this helps.

zhoxing-ms replied:
Got it, thanks~

panchagnula reviewed (commented):
Since az webapp up tests are all live remove all the yaml recording from the PR. these don't need to be included.

StrawnSC commented:
> Since az webapp up tests are all live remove all the yaml recording from the PR. these don't need to be included.

Just removed the yaml test recordings

zhoxing-ms commented:
@StrawnSC Could you please fix the CI issues?

panchagnula commented:
@StrawnSC , can you check the errors & get the build to green so we can get this change in the current release?
 FAILED src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_webapp_up_commands.py::WebAppUpE2ETests::test_webapp_up_invalid_name
FAILED src/azure-cli/azure/cli/command_modules/appservice/tests/latest/test_webapp_up_commands.py::WebAppUpE2ETests::test_webapp_up_name_exists_not_in_subscription
============= 2 failed, 231 passed, 33 skipped in 92.25s (0:01:32) =============

panchagnula reviewed (approved):
approved
